### PR TITLE
Improve printing of Complex

### PR DESCRIPTION
**Pull request by Sumith1896**
Fix to https://github.com/sympy/symengine/issues/566
Before
```
p/q + r/s*I
```
After
```
p/q + r*I/s
```
cc @certik @shivamvats @isuruf

### CODE REVIEW

File: symengine/tests/printing/test_printing.cpp [hunk 1]
```diff
@@ -97,35 +97,34 @@ TEST_CASE("test_printing(): printing", "[printing]")
...
     r1 = mul(c1, x);
     r2 = mul(c2, x);
-    REQUIRE(c1->__str__() == "5/7*I");
-    REQUIRE(c2->__str__() == "-5/7*I");
-    REQUIRE(r1->__str__() == "5/7*I*x");
-    REQUIRE(r2->__str__() == "-5/7*I*x");
+    REQUIRE(c1->__str__() == "5*I/7");
+    REQUIRE(c2->__str__() == "-5*I/7");
+    //REQUIRE(r1->__str__() == "5/7*I*x");
+    //REQUIRE(r2->__str__() == "-5/7*I*x");
```
Comment (Sumith1896) [inline symengine/tests/printing/test_printing.cpp]: I initially thought that the printer class takes care of this.
These need to be fixed, any suggestions?

**Comment by certik** [comment]
I don't mind adding parentheses, but so far I think I prefer `p/q + r/s*I` to `p/q + r*I/s`. Let me check how tools like Mathematica print this.

**Comment by Sumith1896** [comment]
Cool, let me know.
I'll keep this on hold till then

**Comment by Sumith1896** [comment]
I happened to get my hands on Mathematica,
```
In[8]:= 1 + 2*I                                                                 

Out[8]= 1 + 2 I

In[9]:= p/q + r/s*I                                                             

        p   I r
Out[9]= - + ---
        q    s

````

**Comment by Sumith1896** [comment]
@certik What are your views on this?

I now feel keeping it as before is better!

**Comment by Sumith1896** [comment]
@certik ping

**Comment by certik** [comment]
I don't really have strong views on this either way. We can have both, you can just add it as a new printer and override that one method.

As to Mathematica, you need to force it to print in just one line, not pretty printing.

File: symengine/printer.h [hunk 2]
```diff
@@ -59,7 +59,7 @@ class Precedence : public BaseVisitor<Precedence> {
             if(x.imaginary_ == 1) {
                 precedence = PrecedenceEnum::Atom;
             } else {
-                precedence = PrecedenceEnum::Mul;
+                precedence = PrecedenceEnum::Add;
```
Comment (certik) [inline symengine/printer.h]: Is this a bug in the `Precedence` class? What is the rationale behind this change?

Comment (Sumith1896) [inline symengine/printer.h]: This case is `real` part is `0` and `imaginary` part is `not 1`. Shouldn't it be `Mul`?
/cc @isuruf

Comment (isuruf) [inline symengine/printer.h]: It should be `PrecedenceEnum::Mul`. For example `2*I` which is a `Complex` number whose printing behaves like a `Mul`.

Comment (Sumith1896) [inline symengine/printer.h]: Oh wait! Sorry. It was `Mul` before. Ignore my comment :)

**Comment by Sumith1896** [comment]
Closing this for now. Not really inclined to change.
